odoo/addons/base/models/ir_actions_report.py

```diff
@@ -499,9 +499,13 @@ def barcode(self, barcode_type, value, width=600, height=100, humanreadable=0, q
                 value = '0%s' % value
         try:
             width, height, humanreadable, quiet = int(width), int(height), bool(int(humanreadable)), bool(int(quiet))
+            bar_border = 4
+            if barcode_type == 'QR' and quiet:
+                bar_border = 0
+
             barcode = createBarcodeDrawing(
                 barcode_type, value=value, format='png', width=width, height=height,
-                humanReadable=humanreadable, quiet=quiet
+                humanReadable=humanreadable, quiet=quiet, barBorder=bar_border
```

Comment:
@KangOl @mart-e @smetl  your opinion on those few line is welcome. We need to be able to print a QR without the surrounding blank space to fix the `l10n_ch` QR code. Otherwise it's hard to make it 46mm x 46mm which is a requirement from the spec.

`quiet` is not supported by "QR" type. And is simlpy ignored.
But we can use `barBorder`.

You can check that at:
https://bitbucket.org/rptlab/reportlab/src/61ba11e7d1430641d97cbce044da6c8828ebac32/src/reportlab/graphics/barcode/qr.py#lines-86

Or do you prefer that we add a new keyword to `barcode`, my intent is to avoid changing signature on stable.

Reply:
BTW it would be a great improvement to also implement `barVersion`
https://bitbucket.org/rptlab/reportlab/src/61ba11e7d1430641d97cbce044da6c8828ebac32/src/reportlab/graphics/barcode/qr.py#lines-88

A using a higher version pushes back the limit of characters.